test/build-images.sh

```diff
@@ -18,40 +18,51 @@ set -ex
 
 IMAGES_BUILDING=false
 
+function has_images_been_built {
+  local images=$(gcloud container images list --repository=${GCR_IMAGE_BASE_DIR})
+  local result=1
+  echo "$images" | grep api-server && \
+  echo "$images" | grep frontend && \
+  echo "$images" | grep scheduledworkflow && \
+  echo "$images" | grep persistenceagent && \
+  echo "$images" | grep viewer-crd-controller && \
+  echo "$images" | grep inverse-proxy-agent && \
+  echo "$images" | grep metadata-writer && \
+  echo "$images" | grep cache-server && \
+  echo "$images" | grep cache-deployer && \
+  echo "$images" | grep visualization-server && result=0
+  return $result
+}
+
 # Image caching can be turned off by setting $DISABLE_IMAGE_CACHING env flag.
 # Note that GCR_IMAGE_BASE_DIR contains commit hash, so whenever there's a code
 # change, we won't use caches for sure.
-BUILT_IMAGES=$(gcloud container images list --repository=${GCR_IMAGE_BASE_DIR})
 if
-  test -z "$DISABLE_IMAGE_CACHING" && \
-  echo "$BUILT_IMAGES" | grep api-server && \
-  echo "$BUILT_IMAGES" | grep frontend && \
-  echo "$BUILT_IMAGES" | grep scheduledworkflow && \
-  echo "$BUILT_IMAGES" | grep persistenceagent && \
-  echo "$BUILT_IMAGES" | grep viewer-crd-controller && \
-  echo "$BUILT_IMAGES" | grep inverse-proxy-agent && \
-  echo "$BUILT_IMAGES" | grep metadata-writer && \
-  echo "$BUILT_IMAGES" | grep cache-server && \
-  echo "$BUILT_IMAGES" | grep cache-deployer && \
-  echo "$BUILT_IMAGES" | grep visualization-server;
+  test -z "$DISABLE_IMAGE_CACHING" && has_images_been_built
 then
   echo "docker images for api-server, frontend, scheduledworkflow, \
     persistenceagent, viewer-crd-controller, inverse-proxy-agent, metadata-writer, cache-server, \
     cache-deployer and visualization-server are already built in ${GCR_IMAGE_BASE_DIR}."
 else
-  echo "submitting cloud build to build docker images for commit ${COMMIT_SHA}..."
-  IMAGES_BUILDING=true
-  CLOUD_BUILD_COMMON_ARGS=(. --async --format='value(id)' --substitutions=_GCR_BASE=${GCR_IMAGE_BASE_DIR})
-  # Split into two tasks because api_server builds slowly, use a separate task
-  # to make it faster.
-  BUILD_ID_API_SERVER=$(gcloud builds submit ${CLOUD_BUILD_COMMON_ARGS[@]} \
```

Comment:
It's too complex to keep track of both cloudbuild tasks, I merged both tasks to one.

Reply:
I figured out how to deal with two builds.